setup/src/Magento/Setup/Module/Di/Code/Scanner/PhpScanner.php

```diff
@@ -197,16 +197,31 @@ public function collectEntities(array $files)
      */
     protected function _fetchNamespace($tokenIterator, $count, $tokens)
     {
-        $namespaceParts = [];
-        for ($tokenOffset = $tokenIterator + 1; $tokenOffset < $count; ++$tokenOffset) {
-            if ($tokens[$tokenOffset][0] === T_STRING) {
-                $namespaceParts[] = "\\";
-                $namespaceParts[] = $tokens[$tokenOffset][1];
-            } elseif ($tokens[$tokenOffset] === '{' || $tokens[$tokenOffset] === ';') {
+        if (version_compare(PHP_VERSION, '8.0') < 0) {
+            $namespaceParts = [];
+            for ($tokenOffset = $tokenIterator + 1; $tokenOffset < $count; ++$tokenOffset) {
+                if ($tokens[$tokenOffset][0] === T_STRING) {
+                    $namespaceParts[] = "\\";
+                    $namespaceParts[] = $tokens[$tokenOffset][1];
+                } elseif ($tokens[$tokenOffset] === '{' || $tokens[$tokenOffset] === ';') {
+                    break;
+                }
+            }
+            return join('', $namespaceParts);
+        }
+
+        $namespace = '';
+        for ($tokenOffset = ($tokenIterator + 1); $tokenOffset < $count; ++$tokenOffset) {
+            if ($tokens[$tokenOffset][0] === T_NAME_QUALIFIED
+                || $tokens[$tokenOffset][0] === T_NAME_FULLY_QUALIFIED
```

Comment:
Looks like there should be different cases for T_NAME_QUALIFIED and T_NAME_FULLY_QUALIFIED.
Example: 
https://github.com/Codeception/Codeception/pull/6000/files#diff-7c1158ad3505bce7e9cd0901e6d103a4a6eb81e7f6739c520b28df7d0ed4feb3R149

Reply:
I've double-checked this piece of code and T_NAME_FULLY_QUALIFIED is not needed here, as in this particular place there is no situation when fully qualified namespace can occur. so I removed it from the condition